### DISCUSSION TITLE
Special case for `classmethod` and `staticmethod`

### DISCUSSION THREAD

**Discussion by ashwinvis** [pradyunsg/furo#310] [Potential Issue] [answered]
The `def` keyword appearing in front of all methods and functions is a very bold, but nice touch. This change was made on 8e5f49940eab0c93e5c762a43f05fec215b190b5. However I noticed that classmethod and staticmethod appear as follows, which feel syntactically wrong:

![image](https://user-images.githubusercontent.com/9155111/142200946-32f73166-777c-430e-8bdf-34c76e951ee7.png)

![image](https://user-images.githubusercontent.com/9155111/142200906-621b7f9f-cf31-4e90-9320-e51c34bf323c.png)

Screenshot from https://snek5000.readthedocs.io/en/latest/_generated/snek5000.output.base.html

**Comment by pradyunsg** [comment]
Hah, interesting. There's absolutely no way for me to accomodate for that with the markup that Sphinx produces.

**Comment by pradyunsg** [comment]
Hmm... Would this be reasonable?

<img width="719" alt="Screenshot 2021-11-17 at 19 10 39" src="https://user-images.githubusercontent.com/3275593/142266802-b645a929-d0ac-4f7a-acea-f1be5af9c4fc.png">

**Reply by pradyunsg** [reply]
My other option is to drop the `def `, which will make me a bit sad. :(

**Comment by ashwinvis** [comment]
Just my 2 cents here,

1. Ideally it should be done as a sphinx extension and have a toggle in the sphinx `conf.py`. With such a opinionated change, some would like it and some won't. Who knows, maybe someone one start would another thread that the `self` parameter is missing from the method's argument list, in the future.
1. To the user the syntax is meaningless, and often the docs caters to the user. But *if* you have to present the syntax, then it should look correct.

**Comment by pradyunsg** [comment, marked as answer]
This was resolved by dropping the `def `.
